File: uncertainty/matlab/uqCh7/spring_function.m
  

  function J = spring_function(c,m,k,obs,t)

  num = sqrt(4*m*k - c^2);
  den = 2*m;
  y = 2*exp((-c/den)*t).*cos((num/den)*t);
  J = (y-obs)*((y-obs)');

File: uncertainty/python/ch7/spring_function.py
def spring_function(c,m,k,obs,t):
   
   import math
   import numpy as np

   num = math.sqrt(4*m*k - c**2);
   den = 2*m;
   y = 2*np.multiply(np.exp((-c/den)*t),np.cos((num/den)*t));
   J = np.inner((y-obs),((y-obs).T));
   return J




